scipy/linalg/matfuncs.py

```diff
@@ -668,3 +669,48 @@ def rounded_sign(x):
         return S0
     else:
         return S0, errest
+
+
+def khatri_rao(a, b):
+    """
+    Khatri-rao product
+
+    Parameters
+    ----------
+    a:  (I, K) ndarray
+        Input array
+    b:  (J, K) ndarray
+        Input array
+
+    Returns
+    -------
+    C:  ((IJ), K) ndarray
+        Khatri-rao product of 'a' and 'b'.
+
+    Notes
+    -----
+    Mathematically:
+
+    A * B = (A_{ij}  \bigotimes B_{ij})_{ij}
+
+    Examples
+    --------
+    >>> from numpy import array
+    >>> from scipy.linalg import khatri_rao
+    >>> khatri_rao(array([[1, 2, 3], [4, 5, 6]]),
+    ...            array([[3, 4, 5], [6, 7, 8], [2, 3, 9]]))
+    array([[ 3,  8, 15],
+           [ 6, 14, 24],
+           [ 2,  6, 27],
+           [12, 20, 30],
+           [24, 35, 48],
+           [ 8, 15, 54]])
+
+    """
+    try:
```

Comment:
Don't use try-except here, that's hardly ever the right pattern to use. Instead:
```
if not a.shape[1] == b.shape[1]:
    raise ValueError("The number of columns for both arrays should be equal.")
```

Also, you probably should check above it that input arrays are 2-D (e.g. with `if a.ndim == 2`) and raise a `ValueError` as well for unhandled shapes.

Reply:
ok